ui/src/worker/jobs/timeSeriesToTableGraph.ts

```diff
@@ -50,6 +50,20 @@ export const timeSeriesToTableGraphWork = (
     return {data: metaQuerySeries, sortedLabels, influxQLQueryType: queryType}
   }
 
+  // #5347 rename columns of the same name
+  const uniqueLabels: Record<string, boolean> = {}
+  sortedLabels.forEach(l => {
+    if (uniqueLabels[l.label]) {
+      const origLabel = l.label
+      let i = 2
+      while (uniqueLabels[origLabel + '_' + i]) {
```

Comment:
Would much rather use the `reduce` Array prototype than a `forEach` with a `while` loop.

Reply:
ok, the outer loop changed to reduce

Reply:
can we not determine if there is a duplicate with the accumulator and then append `_` + `i` and increment the counter without the `while` loop?

Reply:
The labels might not be unique without a while loop. They are supplied by the user, such a situation can happen, when users open new query tabs and start with an existing query before doing modifications.

Having 3 tabs with (user supplied) queries:
1. `select "bla" from "x"."y"."m1" ...`
1. `select "bla" as "bla_2" from "x"."y"."m1" ... `
1. `select "bla" from "x"."y"."m1" ...`

The labels (without a while loop) would be then:
```
['cpu.bla','cpu.bla_2','cpu.bla_2']
```

The query explorer then crashes on error, the user is asked to post this defect again and he has to reload the page to restore the query explorer to a valid state. 

I would rather keep the bulletproof while loop therein.